src/modalities/checkpointing/checkpoint_saving_strategies.py

```diff
@@ -17,15 +15,34 @@ def get_checkpoint_instruction(
         evaluation_result: Dict[str, EvaluationResultBatch] | None = None,
```

Comment:
```suggestion
        evaluation_result: Optional[Dict[str, EvaluationResultBatch]] = None,
```
and `from typing import Optional`